src/MagnumPlugins/DdsImporter/DdsImporter.cpp

```diff
@@ -143,6 +285,160 @@ void swizzlePixels(const PixelFormat format, Containers::Array<char>& data) {
     } else CORRADE_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
 }
 
+std::pair<PixelFormat, PixelType> dxgiToGl(DxgiFormat format) {
+    switch(format) {
+        /* RGBA32 Formats */
+        case DxgiFormat::R32G32B32A32Typeless:
+        case DxgiFormat::R32G32B32A32UInt:
+            return {PixelFormat::RGBAInteger, PixelType::UnsignedInt};
+        case DxgiFormat::R32G32B32A32Float:
+            return {PixelFormat::RGBA, PixelType::Float};
+        case DxgiFormat::R32G32B32A32SInt:
+            return {PixelFormat::RGBAInteger, PixelType::Int};
+
+        /* RGB32 Formats */
+        case DxgiFormat::R32G32B32Typeless:
+        case DxgiFormat::R32G32B32UInt:
+            return {PixelFormat::RGBInteger, PixelType::UnsignedInt};
+        case DxgiFormat::R32G32B32Float:
+            return {PixelFormat::RGB, PixelType::Float};
+        case DxgiFormat::R32G32B32SInt:
+            return {PixelFormat::RGBInteger, PixelType::Int};
+
+        /* RGBA16 Formats */
+        case DxgiFormat::R16G16B16A16Typeless:
+        case DxgiFormat::R16G16B16A16UInt:
+            return {PixelFormat::RGBAInteger, PixelType::UnsignedShort};
+        case DxgiFormat::R16G16B16A16Float:
+            return {PixelFormat::RGBA, PixelType::Float};
+        case DxgiFormat::R16G16B16A16UNorm:
+            return {PixelFormat::RGBA, PixelType::UnsignedShort};
+        case DxgiFormat::R16G16B16A16SNorm:
+            return {PixelFormat::RGBA, PixelType::Short};
+        case DxgiFormat::R16G16B16A16SInt:
+            return {PixelFormat::RGBAInteger, PixelType::Short};
+
+        /* RG32 Formats */
+        case DxgiFormat::R32G32Typeless:
+        case DxgiFormat::R32G32UInt:
+            return {PixelFormat::RGInteger, PixelType::UnsignedInt};
+        case DxgiFormat::R32G32Float:
+            return {PixelFormat::RG, PixelType::Float};
+        case DxgiFormat::R32G32SInt:
+            return {PixelFormat::RGInteger, PixelType::Int};
+
+        /* RGBA8 Formats */
+        case DxgiFormat::R8G8B8A8Typeless:
+        case DxgiFormat::R8G8B8A8UInt:
+            return {PixelFormat::RGBAInteger, PixelType::UnsignedByte};
+        case DxgiFormat::R8G8B8A8UNorm:
+            return {PixelFormat::RGBA, PixelType::UnsignedByte};
+        //case DxgiFormat::R8G8B8A8UNormSRGB:
+            //return {PixelFormat::SRGBA, PixelType::Byte}; // TODO: No such thing as PixelFormat::SRGBA
```

Comment:
Hmm. Apparently there's no way to explicitly mark the pixel data as sRGB and in this case it would be just `RGBA`. Numerous sources (e.g. [Apple](https://developer.apple.com/library/content/documentation/3DDrawing/Conceptual/OpenGLES_ProgrammingGuide/AdoptingOpenGLES3/AdoptingOpenGLES3.html) or e.g. the OpenGL ES 3.2 spec, table 8.2 on page 147) say that the matching pixel format for sRGB(Alpha) is just RGB(A). So I guess that settles it.

This is a bit bad, because only input textures that actually *are* in sRGB should be uploaded to sRGB textures. OTOH, almost all 8-bit color data already are in sRGB, unless someone went an extra step and kept them linear.

More info (huh, I apparently went through all of this already in the past): https://github.com/mosra/magnum/issues/125